libraries/Microsoft.Bot.Builder.Dialogs.Rules/Steps/HttpRequest.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Rules.Steps
+{
+    public class HttpRequest : DialogCommand
+    {
+
+        public enum Method {
+            GET,
+            POST
+        }
+
+        public HttpRequest()
+            : base()
+        {
+        }
+
+        protected override string OnComputeId()
+        {
+            return $"HttpRequest[{method}]";
+        }
+
+        public Method method;
+
+        public string url;
+
+        public string responseProperty;
+
+        public Dictionary<string, string> header;
+
+        public JObject body;
```

Comment:
How do people specify a jobject declaratively?

Reply:
Shouldn't this be string?

Reply:
Then you can parse it to jobject when the command is executed.

Reply:
Actually just saw your sample, does it work? Where is the body converted to jobject?

Reply:
> Actually just saw your sample, does it work? Where is the body converted to jobject?

         
 "body": {
            "id": "{dialog.petid}",
            "category": {
              "id": 0,
              "name": "string"
            },
            "name": "{dialog.petname}",
            "photoUrls": [
              "string"
            ],
            "tags": [
              {
                "id": 0,
                "name": "string"
              }
            ],
            "status": "available"
          },

this will convert the dictionary {...} after the 'body' to JObject, I think it's a little bit a hard to type a JSON string format for user so I choose JObject and it works.